Demo/iOS/AICamera/AICamera/paddle_image_recognizer.cpp

```diff
@@ -0,0 +1,321 @@
+                        
+                        ptDst[k] = (unsigned char)((pixel_00 * a1 + pixel_01 * a2) * b1 +
+                                                   (pixel_10 * a1 + pixel_11 * a2) * b2);
+#endif
+                    }
+                }
+            }  // for j
+        }    // for i
+    }
+#endif
+    
+}  // namespace image
+
+static void* read_config(const char* filename, long* size) {
+    FILE* file = fopen(filename, "rb");
+    if (file == NULL) {
+        fprintf(stderr, "Open %s error\n", filename);
+        return NULL;
+    }
+    fseek(file, 0L, SEEK_END);
+    *size = ftell(file);
+    fseek(file, 0L, SEEK_SET);
+    void* buf = malloc(*size);
+    fread(buf, 1, *size, file);
+    fclose(file);
+    return buf;
+}
+
+void ImageRecognizer::init(const char* merged_model_path,
+                           const size_t normed_height,
+                           const size_t normed_width,
+                           const size_t normed_channel,
+                           const std::vector<float>& means) {
+    // Set the normed image size
+    normed_height_ = normed_height;
+    normed_width_ = normed_width;
+    normed_channel_ = normed_channel;
+    
+    // Set means
+    if (!means.empty()) {
+        means_ = means;
+    } else {
+        means_.clear();
+        for (size_t i = 0; i < normed_channel; ++i) {
+            means_.push_back(0.0f);
+        }
+    }
+    
+    // Step 1: Reading merged model.
+    long size;
+    void* buf = read_config(merged_model_path, &size);
+    
+    // Step 2:
+    //    Create a gradient machine for inference.
+    CHECK(paddle_gradient_machine_create_for_inference_with_parameters(
+                                                                       &gradient_machine_, buf, size));
+    
+    free(buf);
+    buf = nullptr;
+}
+
+void ImageRecognizer::preprocess(const unsigned char* pixels,
+                                 float* normed_pixels,
+                                 const size_t height,
+                                 const size_t width,
+                                 const size_t channel) {
+    // if (height == normed_height && width == normed_width && channel ==
+    // normed_channel) {
+    // } else {
+    // }
+    
+#if 0
+    size_t resized_height = 0;
+    size_t resized_width = 0;
+    if (height > width) {
+        resized_width = normed_width;
+        resized_height = normed_height * height / width;
+    } else /* height <= width */ {
+        resized_width = normed_width * width / height;
+        resized_height = normed_height;
+    }
+#else
+    size_t resized_height = normed_height_;
+    size_t resized_width = normed_width_;
+#endif
+    
+    // Bilinear Interpolation Resize
+    unsigned char* resized_pixels = (unsigned char*)malloc(
+                                                           resized_height * resized_width * channel * sizeof(unsigned char));
+    image::resize_hwc(pixels,
```

Comment:
From 228 to 255 is it possible to merge into one function? Convert the hwc format data with char type to chw format data with float type. This can improve image preprocessing performance.